test/EndToEnd/tests/Settings.ps1

```diff
@@ -44,4 +44,27 @@ function Test-GetRepositoryPathFromVsSettings {
         Write-Host 'Testing if $repoPath exists'
 	Assert-True (Test-Path $repoPath)
 	Assert-True (Test-Path (Join-Path $repoPath "elmah.1.2.2"))
+}
+
+function Test-GetMachineWideSettingBaseDirectoryInVSIX {
+    param($context)
+
+	# Arrange
+	$dte = [System.Runtime.InteropServices.Marshal]::GetActiveObject("VisualStudio.DTE")
+	[string]$vsVersionString = $dte.version
+	[string]$machineWideSettingsBaseDirExpected
+	if (($vsVersionString -as [double]) -ne $null) {
+		[double]$vsVersion = $vsVersionString
+		if($vsVersion -le 14) {
+		$machineWideSettingsBaseDirExpected = Join-Path (Get-ChildItem Env:PROGRAMDATA).Value "\NuGet\Config"
+		}
+		else {
+		$machineWideSettingsBaseDirExpected = Join-Path (Get-ChildItem Env:PROGRAMFILES).Value "\NuGet\Config"
+		}
+	}
+	# Act
+	$machineWideSettingsBaseDirActual = [NuGet.Common.NuGetEnvironment]::GetFolderPath([NuGet.Common.NuGetFolderPath]::MachineWideSettingsBaseDirectory)
+
+	# Assert
+	Assert-True [string]::Compare($machineWideSettingsBaseDirExpected, $machineWideSettingsBaseDirActual, $True)
```

Comment:
`Assert-True` and `Assert-False` are optimal for trivial boolean values.  The disadvantage of using `Assert-True` here is that the expected and actual values are not logged.  This makes it impossible to root cause analyze a failure from the log file alone.

Since PowerShell doesn't have function overloading, I think a better solution would be to add an `Assert-StringEqual` function which handles case-sensitive/-insensitive comparisons and logs the expected and actual results.